### PR TITLE
Update SDK & API

### PR DESCRIPTION
**Pull request by PrXenon**
Update to latest  Android SDK und API
minSDK now 23
targetSDK 33

add versionnumber to plugins
chaged the location for external plugins (testing)

### CODE REVIEW

File: mobile/app/build.gradle [hunk 1]
```diff
@@ -24,15 +24,14 @@ def metaFDroid(outputFileName) {
...
 
     defaultConfig {
         applicationId "org.exarhteam.iitc_mobile"
-        minSdkVersion 17
-        targetSdkVersion 29
+        minSdkVersion 23
+        targetSdkVersion 33
         versionCode = getVersionCodeTimeStamps()
-        versionName "0.34.1"
+        versionName "0.40.1"
```
Comment (modos189) [inline mobile/app/build.gradle]: Please remove the version change, since a new number will be assigned when a new version of the IITC is released

File: mobile/app/build.gradle [hunk 2]
```diff
@@ -60,7 +59,7 @@ android {
     buildTypes {
         debug {
             applicationIdSuffix ".debug"
-            versionNameSuffix "-test-$BUILD_DATE"
+            versionNameSuffix "-test-px-$BUILD_DATE"
```
Comment (modos189) [inline mobile/app/build.gradle]: Apparently this is an unnecessary change

File: mobile/app/build.gradle [hunk 3]
```diff
@@ -78,33 +77,35 @@ android {
...
+    lint {
+        abortOnError false
+    }
 
     applicationVariants.all { variant ->
         variant.resValue  "string", "application_id", variant.applicationId
         variant.getAssembleProvider().configure() {
-            it.doLast {
-                metaFDroid(variant.getPackageApplicationProvider().get().outputDirectory)
+            def outputDir = new File("${project.buildDir.absolutePath}/outputs/apk/${variant.flavorName}/${variant.buildType.name}")
```
Comment (modos189) [inline mobile/app/build.gradle:92]: Won't this cause an error when building an aab instead of an apk?

File: mobile/app/src/debug/res/values/app_name.xml [hunk 4]
```diff
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <string name="app_name">IITC Mobile Test</string>
+    <string name="app_name">IITC Mobile</string>
```
Comment (modos189) [inline mobile/app/src/debug/res/values/app_name.xml]: This change is not necessary because the debug version should have the "Test" postfix

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java [hunk 5]
```diff
@@ -483,7 +508,7 @@ protected void onStart() {
             if (findViewById(R.id.imageLoading).getVisibility() == View.GONE) {
                 // enough idle...let's do some work
                 Log.d("resuming...reset idleTimer");
-                mIitcWebView.loadJS("(function(){if(window.idleReset) window.idleReset();})();");
+                mIitcWebView.loadJS("(function(){if(3,,window.idleReset) window.idleReset();})();");
```
Comment (modos189) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java]: Is that a typo?

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java [hunk 6]
```diff
@@ -1253,6 +1278,16 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
                 mUserLocation.onRuntimePermissionsGranted();
             }
         }
+        if (requestCode == 355) {
```
Comment (modos189) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java:1263]: Maybe this code should be deleted?

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_TileManager.java [hunk 7]
```diff
@@ -33,7 +33,7 @@ public WebResourceResponse getTile(String url) throws Exception {
 
         /*
          * now get the file path...example path:
-         * /storage/emulated/0/Android/data/org.exarhteam.iitc_mobile/files/mqcdn.com/tiles/1.0.0/map/18/137397/89580.jpg
+         * /storage/emulated/0/Android/data/org.exarhteam.iitc_mobile_px/files/mqcdn.com/tiles/1.0.0/map/18/137397/89580.jpg
```
Comment (modos189) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_TileManager.java]: No need to change the string

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebView.java [hunk 8]
```diff
@@ -56,9 +56,9 @@ private void iitc_init(final Context c) {
         mSettings.setDomStorageEnabled(true);
         mSettings.setAllowFileAccess(true);
         mSettings.setGeolocationEnabled(true);
-        mSettings.setAppCacheEnabled(true);
+      //  mSettings.setAppCacheEnabled(true);
```
Comment (modos189) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebView.java]: Is this an intentional change?

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebViewClient.java [hunk 9]
```diff
@@ -259,8 +259,12 @@ public WebResourceResponse shouldInterceptRequest(final WebView view, final Stri
 
         final Uri uri = Uri.parse(url);
         if (uri.getHost() != null && uri.getHost().endsWith(DOMAIN) &&
-                ("http".equals(uri.getScheme()) || "https".equals(uri.getScheme())))
+                ("http".equals(uri.getScheme()) || "https".equals(uri.getScheme()))
+        )
+
+
             return mIitc.getFileManager().getResponse(uri);
+        Log.i("File --> + " +uri);
```
Comment (modos189) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebViewClient.java]: The line that's not needed

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/prefs/PluginPreferenceActivity.java [hunk 10]
```diff
@@ -140,11 +141,16 @@ public boolean onOptionsItemSelected(final MenuItem item) {
             case R.id.menu_plugins_add:
                 if (mFileManager.checkWriteStoragePermissionGranted()) {
                     // create the chooser Intent
+
+                    Log.i("ABLAGE2--> "+ Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOCUMENTS) + "/IITC_Mobile/");
```
Comment (modos189) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/prefs/PluginPreferenceActivity.java]: The line that's not needed

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/prefs/PluginPreferenceActivity.java [hunk 11]
```diff
@@ -343,7 +350,7 @@ private void addHeader(final String title, final boolean userPlugin) {
         final Header newHeader = new Header();
         newHeader.title = title;
         newHeader.fragmentArguments = bundle;
-        newHeader.fragment = "org.exarhteam.iitc_mobile.fragments.PluginsFragment";
+        newHeader.fragment = "org.exarhteam.iitc_mobile_px.fragments.PluginsFragment";
```
Comment (modos189) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/prefs/PluginPreferenceActivity.java]: I already wrote above that it is not required

**Review by modos189 — CHANGES_REQUESTED** [review]
Thank you for your work. I have prepared a commentary with corrections

File: mobile/app/build.gradle (hunk 1 above)
Comment (PrXenon) [reply mobile/app/build.gradle]: Ok.. i understand.. will do it

File: mobile/app/build.gradle (hunk 2 above)
Comment (PrXenon) [reply mobile/app/build.gradle]: Is a relict so i can install official version and this test side by side.. 

I change it back

File: mobile/app/build.gradle (hunk 3 above)
Comment (PrXenon) [reply mobile/app/build.gradle:92]: I have to test it

File: mobile/app/src/debug/res/values/app_name.xml (hunk 4 above)
Comment (PrXenon) [reply mobile/app/src/debug/res/values/app_name.xml]: Is a relict so i can install official version and this test side by side

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java (hunk 5 above)
Comment (PrXenon) [reply mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java]: Oh.. my bad..

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java (hunk 6 above)
Comment (PrXenon) [reply mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java:1263]: Maybe.. i will add an TODO
I have trying to add the file permission but there changed so many things with sdk 33

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_TileManager.java (hunk 7 above)
Comment (PrXenon) [reply mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_TileManager.java]: Is a relict so i can install official version and this test side by side

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebView.java (hunk 8 above)
Comment (PrXenon) [reply mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebView.java]: No maybe i tryed something... dont remember

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebViewClient.java (hunk 9 above)
Comment (PrXenon) [reply mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebViewClient.java]: For testing i need it.. because i cannot write to the old path to store the external user plugins.. 

Will also add a TODO here for removing the LOG

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/prefs/PluginPreferenceActivity.java (hunk 10 above)
Comment (PrXenon) [reply mobile/app/src/main/java/org/exarhteam/iitc_mobile/prefs/PluginPreferenceActivity.java]: For testing i need it.. because i cannot write to the old path to store the external user plugins..

Will also add a TODO here for removing the LOG

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/prefs/PluginPreferenceActivity.java (hunk 11 above)
Comment (PrXenon) [reply mobile/app/src/main/java/org/exarhteam/iitc_mobile/prefs/PluginPreferenceActivity.java]: Yep will change it back

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebViewClient.java (hunk 9 above)
Comment (modos189) [reply mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebViewClient.java]: You can keep it, but delete it for PR

File: mobile/app/src/debug/res/values/app_name.xml (hunk 4 above)
Comment (modos189) [reply mobile/app/src/debug/res/values/app_name.xml]: I don't understand what the question is. The release version is "IITC Mobile", the beta version is "IITC Mobile Beta", the builds from PR or in debug mode are "IITC Mobile Test". These are three different versions that you can have installed at the same time.
In any case, you can choose any name to build on your own. But please remove this line in PR

**Comment by modos189** [comment]
Please do not delete `mobile/keystore.properties.example`
This file is not used, but is an example for creating `mobile/keystore.properties`

**Comment by modos189** [comment]
In the new version of Gradle, when building bundle, the line `app = exec_gradle(source, dist_ext)` in build_mobile.py returns path `mobile/app/build/outputs/bundle/debug/app.aab`, as before. But the file `mobile/app/build/outputs/bundle/debug/app-debug.aab` is actually created. The specified file does not exist, so the build crashes with an error. Any idea how to fix this in gradle?

I checked that if I replace `app.aab` with `app-debug.aab` in the build script, the bundle build completes successfully.

**Comment by modos189** [comment]
Ah, I see. Changes in the build script are required. In order for me not to create PR, please make these changes in `build_mobile.py`:

image
